### PR TITLE
gr-uhd: Add filter API functions to the USRP blocks

### PR DESCRIPTION
**Pull request by mbr0wn**
This is a rebased version of https://github.com/gnuradio/gnuradio/pull/2200.

### CODE REVIEW

File: gr-uhd/swig/uhd_swig.i [hunk 1]
```diff
@@ -117,10 +117,40 @@
...
 %include <uhd/stream.hpp>
 
+%include <uhd/types/filters.hpp>
+
+%include stdint.i
+
+// Used for lists of filter taps
+%template(uhd_vector_int16_t) std::vector<int16_t>;
+
+%template(filter_info_base_sptr) boost::shared_ptr<uhd::filter_info_base>;
```
Comment (mbr0wn) [inline gr-uhd/swig/uhd_swig.i]: Dang it -- these are std::shared_ptr in UHD 4.0.

Comment (michaelld) [reply gr-uhd/swig/uhd_swig.i]: LOL Progress! Same issue with different Thrift versions ... at least UHD provides version macros that are reliable and easily usable!

Comment (michaelld) [reply gr-uhd/swig/uhd_swig.i]: I'll wait to review until you get the `shard_ptr` thingy fixed up ;)

Comment (dkozel) [reply gr-uhd/swig/uhd_swig.i]: Martin added a UHD version check that addresses this.

**Pull request merged**